tests/devices/qutrit_mixed/test_qutrit_mixed_sampling.py

```diff
@@ -0,0 +1,743 @@
+
+        samples = _sample_state_jax(state, shots=10, prng_key=jax.random.PRNGKey(12))
+        samples2 = _sample_state_jax(state, shots=10, prng_key=jax.random.PRNGKey(12))
+        samples3 = _sample_state_jax(state, shots=10, prng_key=jax.random.PRNGKey(13))
+
+        assert np.all(samples == samples2)
+        assert not np.allclose(samples, samples3)
+
+    def test_sample_state_custom_rng(self, two_qutrit_state):
+        """Tests that a custom RNG can be used with sample_state."""
+        custom_rng = np.random.default_rng(12345)
+        samples = sample_state(two_qutrit_state, 4, rng=custom_rng)
+        expected = [[0, 2], [1, 0], [2, 1], [1, 2]]
+        assert qml.math.allequal(samples, expected)
+
+    @flaky
+    def test_entangled_qutrit_samples_always_match(self):
+        """Tests that entangled qutrits are always in the same state."""
+        num_samples = 10000
+
+        bell_state_vector = np.array([[1, 0, 0, 0, 1, 0, 0, 0, 1]])
+        bell_state = get_dm_of_state(bell_state_vector, 2, 3)
+
+        samples = sample_state(bell_state, num_samples)
+        assert samples.shape == (num_samples, 2)
+        assert not any(samples[:, 0] ^ samples[:, 1])  # all samples are entangled
+
+        # all samples are approximately equivalently sampled
+        assert np.isclose(
+            math.count_nonzero(samples[:, 0] == 0) / num_samples,
+            1 / 3,
+            atol=APPROX_ATOL,
+        )
+        assert np.isclose(
+            math.count_nonzero(samples[:, 0] == 1) / num_samples,
+            1 / 3,
+            atol=APPROX_ATOL,
+        )
+        assert np.isclose(
+            math.count_nonzero(samples[:, 0] == 2) / num_samples,
+            1 / 3,
+            atol=APPROX_ATOL,
+        )
+
+    @pytest.mark.slow
+    @pytest.mark.parametrize(
+        "num_wires,shuffled_wires",
+        (
+            (8, [7, 5, 3, 1, 6, 0, 4, 2]),
+            (9, [4, 0, 2, 1, 5, 7, 6, 8, 3]),
+        ),
+    )
+    def test_sample_state_many_wires(self, num_wires, shuffled_wires):
+        """Tests that sample_state works as expected with many wires, and with re-ordering."""
+        shots = 10000
+        probs_shape = (QUDIT_DIM,) * num_wires
+        state_shape = probs_shape * 2
+        diagonal = np.arange(1, QUDIT_DIM**num_wires + 1, dtype=np.float64)
+        diagonal = diagonal / np.sum(diagonal)
+        state = np.diag(diagonal).reshape(state_shape)
+
+        ordered_samples = sample_state(state, shots)
+        ordered_probs = samples_to_probs(ordered_samples, num_wires)
+        assert np.allclose(ordered_probs, diagonal, atol=APPROX_ATOL)
+
+        random_samples = sample_state(state, shots, wires=shuffled_wires)
+        random_probs = samples_to_probs(random_samples, num_wires)
+
+        reordered_probs = ordered_probs.reshape(probs_shape).transpose(shuffled_wires).flatten()
+        assert np.allclose(reordered_probs, random_probs, atol=APPROX_ATOL)
+
+
+class TestMeasureWithSamples:
```

Comment:
Test measurements with samples, When comparing to qubit version I added some tests for counts and reduced the number of tests for expval and variance, I removed the tests for probs because I haven't added that yet.

Reply:
I intended to leave it out as it will take a bit more development time, but I can add it if you think its worth adding.

Reply:
Okay let's make sure to add that one to the list, then!